pkg/scheduler/core/spreadconstraint/select_clusters.go

```diff
@@ -21,30 +21,40 @@ import (
 
 	"k8s.io/klog/v2"
 
-	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
 	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
 )
 
 // SelectBestClusters selects the cluster set based the GroupClustersInfo and placement
-func SelectBestClusters(placement *policyv1alpha1.Placement, groupClustersInfo *GroupClustersInfo, needReplicas int32) ([]*clusterv1alpha1.Cluster, error) {
+func SelectBestClusters(placement *policyv1alpha1.Placement, groupClustersInfo *GroupClustersInfo, needReplicas int32) ([]ClusterDetailInfo, error) {
 	if len(placement.SpreadConstraints) == 0 || shouldIgnoreSpreadConstraint(placement) {
-		var clusters []*clusterv1alpha1.Cluster
-		for _, cluster := range groupClustersInfo.Clusters {
-			clusters = append(clusters, cluster.Cluster)
-		}
 		klog.V(4).Infof("Select all clusters")
-		return clusters, nil
+		return groupClustersInfo.Clusters, nil
 	}
 
 	if shouldIgnoreAvailableResource(placement) {
 		needReplicas = InvalidReplicas
 	}
 
-	return selectBestClustersBySpreadConstraints(placement.SpreadConstraints, groupClustersInfo, needReplicas)
+	clusterList, err := selectBestClustersBySpreadConstraints(placement.SpreadConstraints, groupClustersInfo, needReplicas)
+	if err != nil {
+		return nil, err
+	}
+	clusterMap := make(map[string]ClusterDetailInfo)
+	for _, cluster := range groupClustersInfo.Clusters {
+		clusterMap[cluster.Name] = cluster
+	}
+	var clusters []ClusterDetailInfo
+	for _, cluster := range clusterList {
+		if item, exist := clusterMap[cluster.Name]; exist {
+			clusters = append(clusters, item)
+		}
+	}
```

Comment:
Can you remind me why need to organize the returned cluster again?

In other worlds, what's the difference between `clusterList` and `clusters`?